### PR TITLE
feat: user interaction join or create channel

### LINKED ISSUE #80: User interaction: Join or create channel
- [x] Users should create or join channel, try to create channel, if it's already present, join the channel
- [x] Retrieve user list from a channel right after joining an existing one (should enqueue a user interaction)

### PR DESCRIPTION
**Pull request by lauti7**
This PR: #80

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -388,9 +388,50 @@ impl Client {
         }
     }
 
-    pub async fn join_room(&self, room_id: &RoomId) {
+    pub async fn join_room(&self, room_id: &RoomId, for_channel: bool) {
```
Comment (guidota) [inline src/client.rs:391]: nit: use an enum instead of bool

File: src/user.rs [hunk 2]
```diff
@@ -266,6 +270,11 @@ impl User {
         self.client.read_messages(room_id).await;
     }
 
+    async fn get_channel_members(&self, room_id: OwnedRoomId) {
+        log::debug!("user '{}' act => {}", self.localpart, "GET CHANNEL MEMBERS");
+        self.client.get_channel_members(&room_id).await
```
Comment (guidota) [inline src/user.rs:275]: should we do something with the result?

**Review by guidota — APPROVED** [review]
LGTM! :+1:

File: src/user.rs (hunk 2 above)
Comment (lauti7) [reply src/user.rs:275]: IDK. nothing specified I am not sure if it's useful to store it or to do something like that. @moliva what do u think?

File: src/client.rs [hunk 3]
```diff
@@ -388,9 +392,50 @@ impl Client {
...
     }
 
-    pub async fn join_room(&self, room_id: &RoomId) {
+    pub async fn join_room(&self, room_id: &RoomId, room_type: MessageType) {
         let request = JoinRoomRequest::new(room_id);
         self.send_and_notify(request, UserRequest::JoinRoom).await;
+        if let MessageType::Channel = room_type {
+            self.sync_channel
+                .0
+                .send(SyncEvent::GetChannelMembers(room_id.to_owned()))
```
[the remark this answers is not in the document]
Comment (lauti7) [reply src/client.rs]: it'll be enqueued for the next tick, like other events. The user will react to this event in the next tick

File: src/user.rs [hunk 4]
```diff
@@ -356,6 +404,8 @@ fn pick_random_action(probability_to_act: usize, channels_enabled: bool) -> Soci
             SocialAction::LogOut
         } else if rng.gen_ratio(1, 50) && channels_enabled {
             SocialAction::CreateChannel
+        } else if rng.gen_ratio(1, 35) && channels_enabled {
```
[the remark this answers is not in the document]
Comment (lauti7) [reply src/user.rs:420]: you mean: 
`if channels_enabled && rng.gen_ratio(1, 35)`?

[the remark this answers is not in the document]
Comment (moliva) [reply src/user.rs:420]: Rightly so!

**Pull request merged**